src/main/java/frc/robot/subsystems/endEffector/EndEffectorSubsystem.java

```diff
@@ -41,8 +41,8 @@ public Command setVoltage(double voltage) {
     return Commands.runOnce(() -> endEffectorIO.setVoltage(voltage));
   }
 
-  public Command shoot() {
-    return Commands.runOnce(() -> endEffectorIO.setVoltage(-9));
+  public Command shoot(double voltage) {
```

Comment:
Consider adding the parameter of which level you are targeting, and adding a new field in the CoralLevel enum for shoot voltage.